packages/components/src/autocomplete/get-default-use-items.tsx

```diff
@@ -43,9 +48,11 @@ function filterOptions( search, options = [], maxResults = 10 ) {
 	return filtered;
 }
 
-export default function getDefaultUseItems( autocompleter ) {
+export default function getDefaultUseItems(
+	autocompleter: WPCompleter
+): NonNullable< WPCompleter[ 'useItems' ] > {
```

Comment:
`NonNullable needed here because `useItems` is optional and potentially undefined, but this default function's return value should not be.

Reply:
For function returns, I generally find it simpler to rely on inference if possible. (Unless for example we need to guarantee a stable return type for a publicly exported function, which is not the case here.)

```suggestion
) {
return ( filterValue: string ) => {
```

Reply:
I usually do the same. The goal here was to ensure that the return value of `getDefaultUseItems` always had the same shape as the `Autocomplete`'s `useItems` prop, because if `useItems` isn't provided, [this function is called in its place](https://github.com/WordPress/gutenberg/blob/a3f56ce01fe7deddbd5153decdebf0b9dc656c3f/packages/components/src/autocomplete/autocompleter-ui.tsx#L23).

Looking at it now, I don't see a problem with inference, but I worry I'm forgetting something from the original change. @ciampo if I remember right, this return value typing declaration came out of one of our pairing sessions. I don't know if it rings any bells or if you have thoughts one way on the other regarding an explicit return type vs. inference in this case?

Reply:
As Lena said, we should try to let TypeScript infer the return type.

Applying Lena's suggestion would allow TS to infer a return type of  `KeyedOption[][]`, but I see that `WPCompleter[ 'useItems' ]`'s type is a bit stricter than that: the outer array is expected to contain exactly one item (`[ Array< KeyedOption > ]`).

If we need to make sure that `getDefaultUseItems`'s return type is also guaranteeing an outer array with one item, we can use the `as const` assertion to tell TS to narrow the return type as much as possible

```diff
diff --git a/packages/components/src/autocomplete/get-default-use-items.tsx b/packages/components/src/autocomplete/get-default-use-items.tsx
index 71ce5cde9e..cb42c32a79 100644
--- a/packages/components/src/autocomplete/get-default-use-items.tsx
+++ b/packages/components/src/autocomplete/get-default-use-items.tsx
@@ -50,8 +50,8 @@ function filterOptions(
 
 export default function getDefaultUseItems(
 	autocompleter: WPCompleter
-): NonNullable< WPCompleter[ 'useItems' ] > {
-	return ( filterValue ) => {
+) {
+	return ( filterValue: string ) => {
 		const [ items, setItems ] = useState< Array< KeyedOption > >( [] );
 		/*
 		 * We support both synchronous and asynchronous retrieval of completer options
@@ -119,6 +119,6 @@ export default function getDefaultUseItems(
 			};
 		}, [ filterValue ] );
 
-		return [ items ];
+		return [ items ] as const;
 	};
 }
```

With these changes, TS now infers a return type of `(filterValue: string) => readonly [KeyedOption[]]`, which should be correct and in line with `WPCompleter[ 'useItems' ]`'s type

Reply:
Sounds good. Implemented in 3ca9a1dd2f